OoyalaSkinSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java

```diff
@@ -0,0 +1,165 @@
+package com.ooyala.sample.players;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
+
+import com.ooyala.android.OoyalaNotification;
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.skin.OoyalaSkinLayout;
+import com.ooyala.android.skin.OoyalaSkinLayoutController;
+import com.ooyala.android.util.SDCardLogcatOoyalaEventsLogger;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+/**
+ * This class asks permission for WRITE_EXTERNAL_STORAGE. We need it for automation hooks
+ * as we need to write into the SD card and automation will parse this file.
+ */
+public abstract class AbstractHookActivity extends Activity implements Observer, DefaultHardwareBackBtnHandler {
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	String TAG = this.getClass().toString();
+	protected OoyalaSkinLayoutController playerLayoutController;
+
+	SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+
+	protected String embedCode;
+	protected String pcode;
+	protected String domain;
+
+	OoyalaPlayer player;
+	//protected OptimizedOoyalaPlayerLayoutController playerLayoutController;
+	protected OoyalaSkinLayoutController controller;
```

Comment:
remove this variable, you already have another of the same type (`playerLayoutController`).